senderworker/src/main/scala/com/gu/notifications/ec2worker/Configuration.scala

```diff
@@ -0,0 +1,67 @@
+package com.gu.notifications.ec2worker
+
+import com.gu.notifications.worker.delivery.apns.models.ApnsConfig
+import com.typesafe.config.Config
+import db.JdbcConfig
+import com.gu.notifications.worker.delivery.fcm.models.FcmConfig
+import _root_.models.{Android, AndroidBeta, AndroidEdition, Ios, IosEdition, Platform}
+import com.gu.{AppIdentity, AwsIdentity}
+import com.gu.conf.{ConfigurationLoader, SSMConfigurationLocation}
+import com.gu.notifications.worker.ApnsWorkerConfiguration
+import com.gu.notifications.worker.FcmWorkerConfiguration
+
+class ConfigWithPrefix(config: Config, prefix: String) {
+    
+    def getString(key: String): String = config.getString(s"${prefix}.${key}")
+
+    def getBoolean(key: String): Boolean = config.getBoolean(s"${prefix}.${key}")
+
+    def getInt(key: String): Int = config.getInt(s"${prefix}.${key}")
+}
+
+object Configuration {
+
+  def fetchConfiguration(): Config = {
+    val identity = AppIdentity.whoAmI(defaultAppName = "sender-worker")
+    ConfigurationLoader.load(identity) {
+      case AwsIdentity(_, _, stage, _) => SSMConfigurationLocation(s"/notifications/$stage/ec2workers")
+    }
+  }
+
+  def fetchApns(appConfig: Config, platform: Platform): ApnsWorkerConfiguration = {
+    val config = new ConfigWithPrefix(appConfig, platform.toString())
+    ApnsWorkerConfiguration(
+      config.getString("cleaningSqsUrl"),
+      ApnsConfig(
+        teamId = config.getString("apns.teamId"),
+        bundleId = config.getString("apns.bundleId"),
+        keyId = config.getString("apns.keyId"),
+        certificate = config.getString("apns.certificate"),
+        mapiBaseUrl = config.getString("mapi.baseUrl"),
+        sendingToProdServer = config.getBoolean("apns.sendingToProdServer"),
+        dryRun = config.getBoolean("dryrun")
+      ),
+      config.getInt("apns.threadPoolSize")
+    )
+  }
+
+  def fetchFirebase(appConfig: Config, platform: Platform): FcmWorkerConfiguration = {
+    val config = new ConfigWithPrefix(appConfig, platform.toString())
+
+    def getStringList(path: String): List[String] =
+      appConfig.getString(path).split(",").toList
+
+    FcmWorkerConfiguration(
+      config.getString("cleaningSqsUrl"),
+      FcmConfig(
+        serviceAccountKey = config.getString("fcm.serviceAccountKey"),
+        debug = config.getBoolean("fcm.debug"),
+        dryRun = config.getBoolean("dryrun")
+      ),
+      config.getInt("fcm.threadPoolSize"),
+      getStringList("fcm.allowedTopicsForBatchSend")
```

Comment:
It mainly duplicates the logic of building `FcmWorkerConfiguration` and `ApnsWorkerConfiguration` from the configuration, but the `Config` in EC2 sender contains all the configuration of each platform so the logic in lambda sender cannot be reused directly.

Given that we rarely add or remove configuration items and we will switch over to (or dispose of) EC2 sender at the end, maybe it is acceptable as a temporary solution.  But any idea would be great!

Reply:
I think this duplication is fair - like you said this should only exist in this medium-term. After we decide on an infrastructure (lambda or ec2) we won't need both Config files. 

Thank you for pointing this out though, hopefully we won't need to add new remote parameters, but if we do we will need to parse them in 2 places in the codebase (for now at least).